client/components/Editor/plugins/suggestedEdits/resolve.ts

```diff
@@ -16,16 +16,16 @@ const getResolvableRangeForSelection = (
 	const suggestedEditsState = getSuggestedEditsState(state);
 	if (suggestedEditsState) {
 		const { suggestionRanges } = suggestedEditsState;
-		if (selection.from === selection.to) {
-			const suggestionRange = suggestionRanges.find(
-				(range) => range.from <= selection.from && selection.from <= range.to,
-			);
-			if (suggestionRange) {
-				return suggestionRange;
-			}
-			return null;
+		const suggestionRange = suggestionRanges.find((range) => {
+			const selectionStartIntersectsSuggestion =
+				range.from <= selection.from && selection.from <= range.to;
+			const selectionEndIntersectsSuggestion =
+				range.from <= selection.to && selection.to <= range.to;
+			return selectionStartIntersectsSuggestion || selectionEndIntersectsSuggestion;
+		});
+		if (suggestionRange) {
+			return selection.empty ? suggestionRange : selection;
```

Comment:
When an empty selection is passed (or a cursor) we can assume that the first suggestion found will be the only one, so we return that. But when a range of text is selected, we return the actual selection if any suggestions were found inside, so that we can still resolve _all_ the suggestions in that range.

Reply:
lol thats pretty cool

what doe these lines mean?
```
const selectionStartIntersectsSuggestion =
		range.from <= selection.from && selection.from <= range.to;
const selectionEndIntersectsSuggestion =
		range.from <= selection.to && selection.to <= range.to;
return selectionStartIntersectsSuggestion || selectionEndIntersectsSuggestion;
```

is this saying a range can be more than the selection? or i suppose how are `suggestionRanges` made

Reply:
hm yeah i think these are named confusingly and i should revisit this logic

Reply:
actually these are named right but just miss a pretty important case: where the range is entirely included inside the selection but doesn't intersect with the start or end